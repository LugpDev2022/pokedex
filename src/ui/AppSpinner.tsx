import { Spinner } from "react-bootstrap";

interface Props {
  centered: boolean;
  variant?: string;
}

export const AppSpinner = ({ centered, variant }: Props) => {
  if (centered) {
    return (
      <div className="centered-spinner position-absolute top-50 start-50 translate-middle">
        <Spinner animation="border" role="status" variant={variant && variant}>
          <span className="visually-hidden">Loading...</span>
        </Spinner>
      </div>
    );
  }

  return (
    <Spinner animation="border" role="status">
      <span className="visually-hidden">Loading...</span>
    </Spinner>
  );
};
