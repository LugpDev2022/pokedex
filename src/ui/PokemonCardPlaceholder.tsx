import { Card, Col, ListGroup, Placeholder, Row } from "react-bootstrap";
import white from "../assets/images/white.jpg";

export const PokemonCardPlaceholder = () => (
  <Col xs={11} sm={9} md={6} lg={3}>
    <Card className="mt-5">
      <Card.Header>
        <Row>
          <span>
            <Placeholder xs={7} />
          </span>
        </Row>
        <Placeholder xs={2} />
      </Card.Header>
      <Card.Img src={white} />
      <ListGroup variant="flush">
        <ListGroup.Item className="fs-5">
          <Placeholder xs={6} />
        </ListGroup.Item>
        <ListGroup.Item className="fs-5">
          <Placeholder xs={4} />
        </ListGroup.Item>
        <ListGroup.Item className="fs-5">
          <Placeholder xs={5} />
        </ListGroup.Item>
      </ListGroup>
    </Card>
  </Col>
);
