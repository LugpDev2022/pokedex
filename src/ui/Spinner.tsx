import { Spinner } from "react-bootstrap";

interface Props {
  centered: boolean;
}

export const AppSpinner = ({ centered }: Props) => {
  if (centered) {
    return (
      <div className="centered-spinner position-absolute top-50 start-50 translate-middle">
        <Spinner animation="border" role="status">
          <span className="visually-hidden">Loading...</span>
        </Spinner>
      </div>
    );
  }

  return (
    <Spinner animation="border" role="status">
      <span className="visually-hidden">Loading...</span>
    </Spinner>
  );
};
