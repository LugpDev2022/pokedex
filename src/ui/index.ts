export * from "./PokemonCardPlaceholder";
export * from "./Spinner";
