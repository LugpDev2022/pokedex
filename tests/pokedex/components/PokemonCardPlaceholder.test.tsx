import React from "react";
import { render } from "@testing-library/react";
import { PokemonCardPlaceholder } from "../../../src/ui/PokemonCardPlaceholder";

describe("tests on <PokemonCardPlaceholder />", () => {
  test("should match snapshot", () => {
    const { container } = render(<PokemonCardPlaceholder />);
    expect(container).toMatchSnapshot();
  });
});
