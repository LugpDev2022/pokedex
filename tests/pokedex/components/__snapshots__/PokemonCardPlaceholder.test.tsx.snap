// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <PokemonCardPlaceholder /> should match snapshot 1`] = `
<div>
  <div
    class="col-lg-3 col-md-6 col-sm-9 col-11"
  >
    <div
      class="mt-5 card"
    >
      <div
        class="card-header"
      >
        <div
          class="row"
        >
          <span>
            <span
              class="col-7 placeholder"
            />
          </span>
        </div>
        <span
          class="col-2 placeholder"
        />
      </div>
      <img
        class="card-img"
        src=""
      />
      <div
        class="card-img-overlay"
      >
        <div
          class="centered-spinner position-absolute top-50 start-50 translate-middle"
        >
          <div
            class="spinner-border text-muted"
            role="status"
          >
            <span
              class="visually-hidden"
            >
              Loading...
            </span>
          </div>
        </div>
      </div>
      <div
        class="list-group list-group-flush"
      >
        <div
          class="fs-5 list-group-item"
        >
          <span
            class="col-6 placeholder"
          />
        </div>
        <div
          class="fs-5 list-group-item"
        >
          <span
            class="col-4 placeholder"
          />
        </div>
        <div
          class="fs-5 list-group-item"
        >
          <span
            class="col-5 placeholder"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
